tests/rptest/tests/upgrade_with_workload.py

```diff
@@ -42,24 +45,32 @@ def upgrade_with_workload(redpanda: RedpandaService, initial_version,
 
         # Upgrade one node and send RPCs in both directions.
         installer.install([node0], RedpandaInstaller.HEAD)
-        redpanda.restart_nodes([node0])
+        redpanda.rolling_restart_nodes([node0],
```

Comment:
Based on the commit message the true problem is that we need to improve health checks in the test infra instead of using rolling restart as a workaround. If that's true, then we should figure out those improvements instead.

I recognize that this is out of the scope of this PR, but maybe we should start a discussion over this.

Reply:
Partially, though in general, waiting for the cluster to become healthy seems reasonable in the context of immediately running RPCs and workloads.